### ISSUE TITLE
Filtering results based on path

### ISSUE THREAD

**Issue by lukewhitehouse** [humanseelabs/gatsby-plugin-lunr#13]
Hi there, I'm wondering if there's any official support for the filtering of pages to be included in search results? Currently I've got a list of posts that all have the following path: `/posts/post-name`, and then I have other random URLs for other areas of my site: `/about`, `/contact`, etc. These are all coming through Markdown and so are being added to the search results, which is less than ideal.

My first thought was to utilise the languages `filterNodes` method in `gatsby-config.js` with something such as:

```
filterNodes: node => node.path !== undefined ? node.path.includes('/posts') : false;
```

When logging out `node.path` I found that a number of nodes didn't have a `path`. Not sure where they were coming from but thats the reason for the conditional. Then it's to check for if the `path` string includes `/posts`, which returns true.

Unfortunately when using this it doesn't seem to work and instead returns an empty index in `search_index.json`:

```
{"en":{"index":{"version":"2.3.5","fields":["title","excerpt","path","content","datetime","category","tags"],"fieldVectors":[],"invertedIndex":[],"pipeline":["stemmer"]},"store":{}}}
```

Any ideas? Here's my full plugin config:

```

  resolve: 'gatsby-plugin-lunr',
  options: {
    languages: [
      {
        name: 'en',
        filterNodes: node => node.path !== undefined ? node.path.includes('/posts') : false;
      }
    ],
    fields: [
      { name: 'title', store: true, attributes: { boost: 20 } },
      { name: 'excerpt', store: true, attributes: { boost: 5 } },
      { name: 'path', store: true },
      { name: 'content' },
      { name: 'datetime', store: true },
      { name: 'category', store: true },
      { name: 'tags', store: true }
    ],
    resolvers: {
      MarkdownRemark: {
        title: node => node.frontmatter.title,
        excerpt: node => node.frontmatter.excerpt,
        path: node => node.frontmatter.path,
        content: node => node.rawMarkdownBody,
        datetime: node => node.frontmatter.datetime,
        category: node => node.frontmatter.category,
        tags: node => node.frontmatter.tags
      }
    },
    filename: 'search_index.json'
  }
}
```

**Comment by deemaagog** [comment]
Hi Luke! It seems you missed 'frontmatter' in your condition. Try something like this:
```javascript
filterNodes: node => node.frontmatter && node.frontmatter.path && node.frontmatter.path.includes('/posts')

**Comment by lukewhitehouse** [comment]
@deemaagog thanks! I did think I tried that but it must have been an earlier version that wasn't quite right.

Interestingly though, if you log out `node.path` it does come back with the path, and when you log out the result of the `includes()` function, each post logs out `true`, yet it doesn't seem to bring back the results in your `search_index.json`. Weird, but that fixes my issue, thanks again! :)
